nstar/src/main.rs

```diff
@@ -12,306 +12,198 @@
-                        "No npk found at {}/{}",
-                        pwd.to_string_lossy(),
-                        npk.to_string_lossy()
-                    )));
-                }
+        }
+        Northstar::Install { npk, repo_id } => {
+            let npk = Path::new(&npk).canonicalize()?;
+            return match client.install(&npk, repo_id).await {
+                Ok(()) => Ok(Response::Ok(())),
+                Err(e) => Ok(Response::Err(api::model::Error::Npk(e.to_string()))),
+            };
+        }
+        Northstar::Uninstall { name, version } => {
+            Request::Uninstall(name.to_owned(), version.clone())
+        }
+        Northstar::Shutdown => Request::Shutdown,
+    };
 
-                let repo_id = if let Some(repository) = split.next() {
-                    repository
-                } else {
-                    return Ok(Some("Missing repository argument".into()));
-                };
-                return match client.install(npk, repo_id).await {
-                    Err(Error::Api(api::model::Error::RepositoryIdUnknown(id, known))) => Ok(Some(
-                        format!("No such repository id: {}, available: {:?}", id, known),
-                    )),
-                    Err(x) => Err(x.into()),
-                    Ok(()) => Ok(None),
-                };
-            }
-            "uninstall" => {
-                let name = if let Some(name) = split.next() {
-                    name
-                } else {
-                    return Ok(Some("Missing container name".into()));
-                };
-                let version = if let Some(version) = split.next() {
-                    match Version::parse(version) {
-                        Ok(version) => version,
-                        Err(e) => return Ok(Some(format!("Invalid version: {}", e))),
-                    }
-                } else {
-                    return Ok(Some("Missing version".into()));
-                };
-                client.uninstall(name, &version).await?;
-            }
-            "shutdown" => {
-                client.shutdown().await?;
-            }
-            c => return Ok(Some(format!("Unimplemented command \"{}\"", c))),
+    client
+        .request(request)
+        .await
+        .context("Failed to get a response")
+}
+
+async fn print_response<W: Write>(mut output: &mut W, response: Response) -> Result<()> {
+    match response {
+        Response::Containers(cs) => pretty::containers(&mut output, &cs),
+        Response::Repositories(rs) => pretty::repositories(&mut output, &rs),
+        Response::Ok(()) => {
+            writeln!(output, "{}", "✔ success".green()).context("Failed to write to stdout")
+        }
+        Response::Err(e) => {
+            writeln!(output, "{}: {:?}", "✗ failed".red(), e).context("Failed to write to stdout")
         }
     }
-    Ok(None)
+}
+
+async fn create_client(host: &str) -> Result<api::client::Client> {
```

Comment:
What's the need for the timeout? There's a TCP connect timeout anyway.

Reply:
I'm not sure, this was simply moved from the current version, will remove it :+1: